benchmark/cajita/Cajita_HaloPerformance.cpp

```diff
@@ -0,0 +1,244 @@
+                                         Cajita::Cell() );
+        auto array =
+            Cajita::createArray<double, device_type>( "array", layout );
+
+        // Assign the owned cells a value of 1 and ghosted 0.
+        Cajita::ArrayOp::assign( *array, 0.0, Cajita::Ghost() );
+        Cajita::ArrayOp::assign( *array, 1.0, Cajita::Own() );
+
+        // create host mirror view
+        auto array_view = array->view();
+
+        for ( int t = 0; t < num_run; ++t )
+        {
+
+            // create halo
+            halo_create_timer.start( halo_width );
+            auto halo =
+                createHalo( *array, Cajita::NodeHaloPattern<3>(), halo_width );
+            halo_create_timer.stop( halo_width );
+
+            // gather
+            halo_gather_timer.start( halo_width );
+            halo->gather( exec_space(), *array );
+            halo_gather_timer.stop( halo_width );
+
+            // scatter
+            halo_scatter_timer.start( halo_width );
+            halo->scatter( exec_space(), Cajita::ScatterReduce::Sum(), *array );
+            halo_scatter_timer.stop( halo_width );
+        }
+    }
+
+    // Output results
+    outputResults( stream, "halo_width", problem_halo_width, halo_create_timer,
+                   comm );
+    outputResults( stream, "halo_width", problem_halo_width, halo_gather_timer,
+                   comm );
+    outputResults( stream, "halo_width", problem_halo_width, halo_scatter_timer,
+                   comm );
+    stream << std::flush;
+}
+
+//---------------------------------------------------------------------------//
+// main
+int main( int argc, char* argv[] )
+{
+    // Initialize environment
+    MPI_Init( &argc, &argv );
+    Kokkos::initialize( argc, argv );
+
+    // Check arguments.
+    if ( argc < 2 )
+        throw std::runtime_error( "Incorrect number of arguments. \n \
+             First argument - file name for output \n \
+             Optional second argument - run size (small or large) \n \
+             \n \
+             Example: \n \
+             $/: ./GridHaloPerformance test_results.txt large\n" );
+
+    // Get the name of the output file.
+    std::string filename = argv[1];
+
+    // Define run sizes.
+    std::string run_type = "";
+    if ( argc > 2 )
+        run_type = argv[2];
+    std::vector<int> problem_global_cells_per_dim = { 16 };
+    std::vector<int> problem_halo_width = { 1, 2 };
+    if ( run_type == "large" )
+    {
+        problem_global_cells_per_dim = { 16, 32, 64, 128 };
+        problem_halo_width = { 1, 2, 3, 4, 5 };
+    }
+
+    // Barier before continuing.
+    MPI_Barrier( MPI_COMM_WORLD );
+
+    // Get comm rank;
+    int comm_rank;
+    MPI_Comm_rank( MPI_COMM_WORLD, &comm_rank );
+
+    // Get comm size;
+    int comm_size;
+    MPI_Comm_size( MPI_COMM_WORLD, &comm_size );
+
+    // Get Cartesian comm
+    std::array<int, 3> ranks_per_dim = {};
+    MPI_Dims_create( comm_size, 3, ranks_per_dim.data() );
```

Comment:
Simpler to create the partitioner here, extract the `ranks_per_dim` to print, then pass it to the function

Reply:
done

Reply:
Let's use `DimBlockPartitioner` and `ranksPerDimension` instead

Reply:
done